Account switch @integration test sometimes fails

### Expected behaviour

Tests should not fail randomly

### Actual behaviour
This test sometimes fails

```
FAILED TESTS:
  @integration: Account switch
    Scenario: should allow to remove a saved account
      ✖ Then I should see 2 instances of "saved account card"
        HeadlessChrome 0.0.0 (Linux 0.0.0)
      AssertionError: expected { length: 3 } to have a length of 2 but got 3
```
https://jenkins.lisk.io/job/lisk-wallet-private/job/PR-162/19/testReport/(root)/@integration_%20Account%20switch/_integration__Account_switch_Scenario__should_allow_to_remove_a_saved_account_Then_I_should_see_2_instances_of__saved_account_card_/

### Steps to reproduce

Run tests repeatedly.

--- a/src/components/savedAccounts/savedAccounts.js
+++ b/src/components/savedAccounts/savedAccounts.js
@@ -41,7 +41,10 @@ class SavedAccounts extends React.Component {
   }
 
   isSelectedForRemove(account) {
-    return account === this.state.accountSelectedForRemove;
+    const { publicKey, network, address } = this.state.accountSelectedForRemove || {};
+    return (account.publicKey === publicKey &&
+      account.network === network &&
+      account.address === address);
   }
 
   handleRemove(account, e) {

